### PR TITLE
Pthread cleanup

### PR DESCRIPTION
**Pull request by rubenk**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1603,45 +1603,21 @@ if test "x$with_libhal" = "xyes"; then
...
-		with_libpthread="yes"
-	else
-		if test "x$withval" = "xno"
-		then
-			with_libpthread="no (disabled)"
-		fi
-	fi
-], [with_libpthread="yes"])
-
 AC_SUBST([PTHREAD_LIBS])
```
Comment (tokkee) [inline configure.ac]: Nit: While it works either way, I think it's cleaner to have AC_SUBST() after defining and populating the underlying variable.

File: configure.ac [hunk 2]
```diff
@@ -1603,45 +1603,21 @@ if test "x$with_libhal" = "xyes"; then
...
-else
-	collect_pthread=0
-fi
-AC_DEFINE_UNQUOTED(HAVE_LIBPTHREAD, [$collect_pthread],
-	[Wether or not to use pthread (POSIX threads) library])
-AM_CONDITIONAL(BUILD_WITH_LIBPTHREAD, test "x$with_libpthread" = "xyes")
-# }}}
+AC_CHECK_HEADERS([pthread.h],
+  [],
+  [AC_MSG_ERROR([pthread.h not found])]
```
Comment (tokkee) [inline configure.ac]: We've got some more '#if HAVE_PTHREAD_H' which could now go away (plugin.h, common.h).

**Comment by tokkee** [comment]
I wonder if the plugins should actually still be linked against pthread. It's not needed on Linux but I seem to recall some weird systems which might need that.

Otherwise LG.

File: configure.ac (hunk 1 above)
Comment (rubenk) [inline configure.ac]: Ok, will fix.

File: configure.ac (hunk 2 above)
Comment (rubenk) [inline configure.ac]: indeed, the include guard makes no sense now since you can't build without pthread.h. Will fix.

**Comment by rubenk** [comment]
I tested Solaris 10 and FreeBSD, it's all good there.

If this breaks on other systems, their runtime loader is broken. Not that that would surprise me ;)
I have no plans to backport this so we'll find out soon enough.

Thanks for the review.

**Pull request merged**